marlon/baseline_models/multiagent/marl_algorithm.py

```diff
@@ -106,6 +103,7 @@ def run_episode(
     obs1 = attacker_agent.env.reset()
 
     if defender_agent:
+        defender_agent.wrapper.on_reset()
```

Comment:
Careful, the way you're using this here implies that `.wrapper` is a property on *all* MarlonAgents, when in fact you only added it to BaselineMarlonAgent. If this were a RandomMarlonAgent, this would throw an exception.

Easy fix: Add the abstract property to MarlonAgent, then implement it in RandomMarlonAgent the same way you did for BaselineMarlonAgent.

Reply:
Will do

Reply:
I cant test this case since we dont have a random defender currently implemented, but I added the property just in case.